### PR TITLE
update: more typing and clean up

### PR DESCRIPTION
**Pull request by EribertoLopez**
Further typing input parameter representations

### CODE REVIEW

File: autoprotocol/instruction.py [hunk 1]
```diff
@@ -554,14 +555,7 @@ class Incubate(Instruction):
...
     WHERE = [
-        "ambient",
-        "warm_30",
-        "warm_35",
-        "warm_37",
-        "cold_4",
-        "cold_20",
-        "cold_80",
-        "cold_196",
+        allowable.strip("_") for allowable in Location.__dict__.get("_member_names_")
```
Comment (sabrinamohabbat) [inline autoprotocol/instruction.py:558]: I noticed "warm_30" is included in this list comprehension but I don't see it in the docstring...not sure if that matters. 
image

**Review by sabrinamohabbat — APPROVED** [review]
LGTM just one question/comment regarding the docstring.

File: autoprotocol/instruction.py (hunk 1 above)
Comment (EribertoLopez) [reply autoprotocol/instruction.py:558]: I just replaced all the strings that I saw in the list. I'll add it to the docstring

**Review by sabrinamohabbat — APPROVED** [review]
LGTM!

File: autoprotocol/types/protocol.py [hunk 2]
```diff
@@ -17,17 +17,19 @@
 AMOUNT_CONCENTRATION = Union[str, Unit]
 CONCENTRATION = Union[str, Unit]
 FLOW_RATE = Union[str, Unit]
-FREQUENCY = Union[str, Unit]
-LENGTH = Union[str, Unit]
+FREQUENCY = Union[str, Unit]  # hertz, rpm
+LENGTH = Union[str, Unit]  # mm
```
Comment (joshhacksthings) [inline autoprotocol/types/protocol.py:21]: nit: make similar to `WAVELENGTH` comment, with expanded unit notation

```suggestion
LENGTH = Union[str, Unit]  # millimeter
```

File: autoprotocol/types/protocol.py [hunk 3]
```diff
@@ -193,26 +200,26 @@ class GelPurifyExtract:
 @dataclass
 class FlowCytometryLaser:
     channels: List[FlowCytometryChannel]
-    excitation: Union[str, Unit] = field(default=None)
-    power: Union[str, Unit] = field(default=None)
+    excitation: WAVELENGTH = field(default=None)
+    power: POWER = field(default=None)
```
Comment (joshhacksthings) [inline autoprotocol/types/protocol.py:204]: do we have to worry about backwards compat where strings were passed to these classes in the past?

**Comment by EribertoLopez** [comment]
closing in favor of https://github.com/autoprotocol/autoprotocol-python/pull/385

**Pull request closed without merging**